### PR TITLE
docs: Update Flutter version in Bare Flame game tutorial

### PR DESCRIPTION
**Pull request by domesticmouse**
# Description

Update the output of `flutter doctor` and the contents of `pubspec.yaml` to reflect 2023.

## Checklist

- [ ] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by domesticmouse** [comment]
@spydon How would you like me to deal with the spell checker failure? I can modify the `flutter doctor` output to not mention problematic IDE?

File: doc/tutorials/bare_flame_game.md [hunk 1]
```diff
@@ -96,13 +101,13 @@ version: 0.0.0
...
-  sdk: ^2.16.0
-  flutter: ^2.10.0
+  sdk: ^3.0.0
+  flutter: ^3.13.0
 
 dependencies:
   flutter:
     sdk: flutter
-  flame: ^--VERSION--
+  flame: ^1.9.1
```
Comment (spydon) [inline doc/tutorials/bare_flame_game.md:110]: This should be kept as it was, so that we don't have to update it all the time

**Comment by spydon** [comment]
> @spydon How would you like me to deal with the spell checker failure? I can modify the `flutter doctor` output to not mention problematic IDE?

You can add it to the [dart_dictionary](https://github.com/flame-engine/flame/blob/main/.github/.cspell/dart_dictionary.txt)

File: doc/tutorials/bare_flame_game.md (hunk 1 above)
Comment (domesticmouse) [reply doc/tutorials/bare_flame_game.md]: reverted

**Comment by spydon** [comment]
Haha what a tricky one, I think you have to add just `intelli` for it to work? :thinking:

**Comment by domesticmouse** [comment]
> Haha what a tricky one, I think you have to add just `intelli` for it to work? 🤔 Or do fix in the verify script

I'm giving `intelli` a try. Colour me amused...

**Pull request merged**